motivaa-backend/motivaa-app/src/test/java/com/motivaa/boundary/UserControllerTest.java

```diff
@@ -86,11 +86,18 @@ void post_Users_MissingEmail() throws Exception {
             request.put("firstName", validFirstName);
             request.put("lastName", validLastName);
 
-            mockMvc.perform(post("/user-apis/users")
+            String responseJson =  mockMvc.perform(post("/user-apis/users")
                     .contentType(MediaType.APPLICATION_JSON)
                     .content(TestUtils.asJsonString(request))
                     .accept(MediaType.APPLICATION_JSON))
-                    .andExpect(status().isBadRequest());
+                    .andExpect(status().isBadRequest())
+                    .andReturn().getResponse().getContentAsString();
+
+            Integer httpCodeFromJson = JsonPath.parse(responseJson).read("$.httpCode");
+            assertEquals(
+                    400,
+                    httpCodeFromJson);
+            TestUtils.assertMissingField(responseJson, "email", "Email is mandatory");
```

Comment:
you could do this inline as part of the mockMvc.perform like this:

.expectBody().jsonPath("$.name").isEqualTo("Yoda");

Reply:
awesome, thanks for the tip! Will adjust them.